File: persistence/src/main/java/io/sparkled/persistence/stage/impl/query/GetStagePropsByStageIdQuery.java
package io.sparkled.persistence.stage.impl.query;

import io.sparkled.model.entity.StageProp;
import io.sparkled.persistence.PersistenceQuery;
import io.sparkled.persistence.QueryFactory;

import java.util.List;

public class GetStagePropsByStageIdQuery implements PersistenceQuery<List<StageProp>> {

    private final int stageId;

    public GetStagePropsByStageIdQuery(int stageId) {
        this.stageId = stageId;
    }

    @Override
    public List<StageProp> perform(QueryFactory queryFactory) {
        return queryFactory
                .selectFrom(qStageProp)
                .where(qStageProp.stageId.eq(stageId))
                .orderBy(qStageProp.displayOrder.asc())
                .fetch();
    }
}

File: persistence/src/main/kotlin/io/sparkled/persistence/stage/impl/query/GetStagePropsByStageIdQuery.kt
package io.sparkled.persistence.stage.impl.query

import io.sparkled.model.entity.QStageProp.Companion.stageProp
import io.sparkled.model.entity.StageProp
import io.sparkled.persistence.PersistenceQuery
import io.sparkled.persistence.QueryFactory

class GetStagePropsByStageIdQuery(private val stageId: Int) : PersistenceQuery<List<StageProp>> {

    override fun perform(queryFactory: QueryFactory): List<StageProp> {
        return queryFactory
            .selectFrom(stageProp)
            .where(stageProp.stageId.eq(stageId))
            .orderBy(stageProp.displayOrder.asc())
            .fetch()
    }
}
